### PR TITLE
Added documentation for building from source (with or without GPU support)

### PR DESCRIPTION
**Pull request by aterenin**
After the discussion in Issue #220, I figured it'd make sense to write a documentation page on building from source. Currently, the information needed for everything to work is scattered in between TensorFlow documentation pages and various issues on this repo.

It would be great to add some information about the binary patch in #245 so that users may incorporate it. That way potential segfaults and unexpected behavior are avoided.

### CODE REVIEW

File: docs/src/build_from_source.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Building TensorFlow from source
+
+Building TensorFlow from source is recommended by Google for maximum performance - on some systems, the difference can be substantial. It will also be required for Mac OS X GPU support for versions later than 1.1. This document describes how to do so.
+
+
+## Step 1: build libtensorflow
+
+To build libtensorflow for TensorFlow.jl, follow the steps here: https://www.tensorflow.org/install/install_sources, except for a few minor modifications.
+
+  * Ignore the step "Install python dependencies".
```
Comment (oxinabox) [inline docs/src/build_from_source.md]: I suggest maybe:

 Ignore the step "Install python dependencies" -- these will not be needed as we are not building  for python. (You do still need to install Bazel though.)

Comment (aterenin) [inline docs/src/build_from_source.md]: Done

**Comment by aterenin** [comment]
All changes should be addressed in latest commit.

**Comment by oxinabox** [comment]
Thank-you.
This is great.

**Pull request merged**